src/lib/io/sol-i2c-impl-linux.c

```diff
@@ -968,8 +968,22 @@ create_device_iter_cb(void *data, const char *dir_path, struct dirent *ent)
     char path[PATH_MAX];
     int r;
     struct stat st;
+#ifdef USE_I2C
+	unsigned int i2c_bus;
+	char *end_ptr;
+#endif
 
     if (strstartswith(ent->d_name, "i2c-")) {
+	#ifdef USE_I2C
+		errno = 0;
+        i2c_bus = strtoul(ent->d_name + strlen("i2c-"), &end_ptr, 0);
+        if (errno || *end_ptr != '\0') {
+			SOL_WRN("Could not get I2C bus number");
+		} else {
+			sol_pin_mux_setup_i2c(i2c_bus);
```

Comment:
You should check the return of this function, because in case a mux recipe was found but went on error during execution, this return will tell you that and you should consider everything in an error situation.

Check 'sol_i2c_open' in sol-i2c-common file, it even closes the bus that was trying to open in case of fault execution of 'sol_pin_mux_setup_i2c', maybe something similar should be done here.